pyvisa_py/tcpip.py

```diff
@@ -769,6 +778,215 @@ def _set_timeout(self, attribute: ResourceAttribute, value: int) -> StatusCode:
         return StatusCode.success
 
 
+class TCPIPInstrVicp(Session):
+    """A VICP Session that uses the network standard library to do the low
+    level communication.
+    """
+
+    # we don't decorate this class with Session.register() because we don't
+    # want it to be registered in the _session_classes array, but we still
+    # need to define session_type to make the set_attribute machinery work.
+    session_type = (constants.InterfaceType.tcpip, "INSTR")
+
+    # Override parsed to take into account the fact that this class is only used
+    # for a specific kind of resource
+    parsed: rname.TCPIPInstr
+
+    @staticmethod
+    def list_resources(wait_time=1.0) -> List[str]:
+        resources = []
+        services = get_services("_lxi._tcp.local.", wait_time=wait_time)
+        for host, properties in services.items():
+            if properties["Manufacturer"].lower().startswith("lecroy"):
+                resources.append(f"TCPIP::{host}::VICP::INSTR")
```

Comment:
I may be misremembering but I had in mind that resource strings for VICP devices were of the form `VICP::{ip address}::INSTR`. I will double check.

Reply:
The way they provided VICP support in NI-VISA struck me as a hack, making VICP support inconsistent with HiSLIP support.  You want to follow their lead?

Reply:
I simply worry that people will get confused if they try to follow their instrument manual. I often wonder how many people read pyvisa docs so I fear that if they need to reach to pyvisa-py doc to figure out why they cannot connect to their VICP device we will get a lot of reports of confused users. 

I agree with you on the principle but I fear it is not practical since Lecroy set the precedent a long time ago.

Reply:
Yes, I see your point.  But I'm also up against a time constraint...I'm retiring in a couple of weeks and won't have access to instruments for testing after that.  I'll see what I can get done...

Reply:
The changes using resource strings of the form `VICP::{ip address}::INSTR` are working using your vicp branch of pyvisa, but I don't know how to tell the continuous integration tools to use that branch.

Reply:
Nevermind, found it: use "@" instead of "#".